### PR TITLE
show a notification after scanned a participant

### PR DESCRIPTION
**Pull request by jiabaow**
I use https://www.npmjs.com/package/react-native-toast-notifications to display notifications after successfully scanning a participant.
Screenshot 2021-11-10 at 17 24 46

### CODE REVIEW

**Review by Xelowak — COMMENTED** [review]
Maybe it could be nice to provide a screenshot in the PR's description, so that we can see how it looks.
You can maybe create a React component for Toast messages if you want to test it independently (since I see that you don't have enough coverage).
Otherwise the code seems nice to me.

File: fe1-web/parts/lao/organizer/RollCallOpened.tsx [hunk 1]
```diff
@@ -29,20 +30,23 @@ const RollCallOpened = () => {
   const route = useRoute();
   const { rollCallID, time } = route.params;
   const navigation = useNavigation();
-  const [, setQrWasScanned] = useState(false);
-  const [attendeesSet, updateAttendeesSet] = useState(new Set<string>());
```
Comment (pierluca) [inline fe1-web/parts/lao/organizer/RollCallOpened.tsx:33]: Careful, a `Set` will automatically avoid de-duplication, whereas a list will allow you to scan 3 times the same key.
You might want to add a test to verify/document this behaviour, actually 😄

File: fe1-web/parts/lao/organizer/__tests__/RollCallOpened.test.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+jest.mock('react-qr-reader', () => function onScan() {
+  const mockToast = useToast();
+  return (
+    mockToast.show('Scanned', {
+      type: 'success',
+      placement: 'top',
+      duration: 4000,
+    })
+  );
+});
```
Comment (pierluca) [inline fe1-web/parts/lao/organizer/__tests__/RollCallOpened.test.tsx:31]: I'm not sure I fully understand what you're trying to do here.

File: fe1-web/parts/lao/organizer/__tests__/RollCallOpened.test.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+describe('RollCallOpened', () => {
```
Comment (pierluca) [inline fe1-web/parts/lao/organizer/__tests__/RollCallOpened.test.tsx:33]: I'd try to add a few tests. Here are a few ideas:
1. Test 1: a toast is shown when a participant is scanned
2. Test 2: when scanning 1 attendee and closing, `onCloseRollCall` is called with 1 PoP token (/ public key)
2. Test 3: when scanning the same attendee 3 times and closing, `onCloseRollCall` is called with 1 PoP token (/ public key)
2. Test 4: when scanning 3 attendees times and closing, `onCloseRollCall` is called with the 3 PoP tokens (/ public keys)

**Review by pierluca — COMMENTED** [review]
I'm not sure this is ready for merge yet. I've tried to provide a few testing ideas as well, so you can get your hands "dirty" with mocks :-)

**Comment by jiabaow** [comment]
> I'm not sure this is ready for merge yet. I've tried to provide a few testing ideas as well, so you can get your hands "dirty" with mocks :-)

Thank you very much for your suggestions! The tests are not finished yet and I am still working on them.

**Comment by jiabaow** [comment]
CI failed because of the import order, but if I fix this issue the tests cannot pass. I think it's because of the dependency.

**Pull request closed without merging**